tests/unit/test_parameters/test_parameter_values.py

```diff
@@ -147,6 +147,10 @@ def test_check_parameter_values(self):
             # since + has other meanings in regex
         with self.assertRaisesRegex(ValueError, "Thermodynamic factor"):
             pybamm.ParameterValues({"1 + dlnf/dlnc": 1})
+        # renamed to "Exchange-current density for lithium-metal electrode [A.m-2]"
```

Comment:
I used `assertRaisesRegex` here to catch the error but it had problems parsing the minus sign (`-`) in the units, so I ended up switching to the usual `assertRaises` instead. Please let me know if is correct.

Reply:
I usually test a substring that works e.g. "Exchange-current density for plating"

Reply:
Okay, I managed to fix that using raw strings in order to keep `regex` happy.

I do have a separate question about how or if the units should be accounted for: because if the units are to be included in the error, `Exchange-current density for plating [A.m-2]` will raise an error as expected, but `Exchange-current density for plating` (without the units) will not